Library/Dnn.PersonaBar.Library/Repository/PersonaBarRepository.cs

```diff
@@ -67,12 +67,12 @@ public PersonaBarMenu GetMenu()
 
         public MenuItem GetMenuItem(string identifier)
         {
-            return GetMenu().AllItems.FirstOrDefault(m => m.Identifier.Equals(identifier, StringComparison.InvariantCultureIgnoreCase));
+            return GetMenu().AllItems.ToList().FirstOrDefault(m => m.Identifier.Equals(identifier, StringComparison.InvariantCultureIgnoreCase));

```

Comment:
I changed the video's permission so that you should be able to watch it now. I have described the behavior in that video. 

To summarize the video, since "AllItems" property references a cache instance, I put ToList method to create a temporary copy of it and apply FirstOrDefault in that temporary variable. Otherwise, FirstOrDefault was working on cache instance's itself which was not safe and might throw this error when cache gets updated.

Reply:
Actually, it will only do a shallow copy which means that only the references will get copied and not the values. The references will still point to the same values so that we should be fine as for performance reasons too.

Reply:
@mitchelsellers I agree with the solution here.

Reply:
This is good enough for now